src/adapters/IDB.js

```diff
@@ -81,6 +87,62 @@ export default class IDB extends BaseAdapter {
   }
 
   /**
+   * Batch operations.
+   *
+   * @param  {Function} fn The operation function, which must return a Promise.
+   * @return {Promise}
+   */
+  batch(fn) {
+    return this.open().then(() => {
+      const errors = [], operations = [];
+      const {transaction, store} = this.prepare("readwrite");
+      const batchObject = {
+        get(id) {
+          if (!id) {
+            return Promise.reject(new Error("Id not provided."));
+          }
+          return new Promise((resolve, reject) => {
+            const request = store.get(id);
+            request.onerror = event => reject(new Error(event.target.error));
+            request.onsuccess = () => resolve(request.result);
+          });
+        }
+      };
+      for (let type of Object.keys(BATCH_STORE_METHODS)) {
+        batchObject[type] = data => {
+          const operation = {type, data};
+          operations.push(operation);
+          return operation;
+        };
+      }
+      const result = fn(batchObject);
+      return Promise.resolve(result)
```

Comment:
Actually there's no point at requiring a Promise to be returned. I'll leave this as optional in the code, especially as I don't clearly see what to do with the returned result value :)

Note: if you have ideas about it, I'm all ears though. I feel like it could be useful, but how?